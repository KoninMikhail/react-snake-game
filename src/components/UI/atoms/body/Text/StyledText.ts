import styled from 'styled-components';
import {
  font_color,
  font_family_body,
  font_size_body,
  font_size_body_desktop,
} from '../../../../../styles/variables';
import { up } from 'styled-breakpoints';
import { ITextProps } from './Text';

export const StyledText = styled.p<ITextProps>`
  font-size: ${font_size_body};
  font-family: ${font_family_body};
  font-weight: normal;
  color: ${font_color};

  ${up('xl')} {
    font-size: ${font_size_body_desktop};
  }
`;
