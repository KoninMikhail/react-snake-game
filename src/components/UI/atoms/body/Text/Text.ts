import styled from 'styled-components';
import {
  font_color,
  font_family_body,
  font_size_body,
  font_size_body_desktop,
} from '../../../../../styles/variables';
import { up } from 'styled-breakpoints';

/**
 * Text
 *
 * @description Use for label content
 *
 * @returns {JSX.Element}
 */
export const Text = styled.p`
  font-size: ${font_size_body};
  font-family: ${font_family_body};
  font-weight: normal;
  color: ${font_color};

  ${up('xl')} {
    font-size: ${font_size_body_desktop};
  }
`;
