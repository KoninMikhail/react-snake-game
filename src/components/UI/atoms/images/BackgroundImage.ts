import styled from 'styled-components';

/**
 * @typedef IBackgroundImageProps
 *
 * @property {string} src - The image src
 * @property {string=} size - The image size (css rules)
 * @property {string=} position - The image position (css rules)
 */
interface IBackgroundImageProps {
  src: string;
  size?: string;
  position?: string;
}

/**
 * BackgroundImage
 *
 * @description Use for add background image
 *
 * @param {IBackgroundImageProps} - Component properties
 * @returns {JSX.Element}
 */
export const BackgroundImage = styled.div<IBackgroundImageProps>`
  background-image: url(${({ src }) => src || ''});
  background-position: ${({ position }) => position || 'center center'};
  background-size: ${({ size }) => size || 'cover'};
  width: inherit;
  height: inherit;
`;
