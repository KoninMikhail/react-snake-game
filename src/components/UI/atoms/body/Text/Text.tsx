import { StyledText } from './StyledText';
import { ElementType } from 'react';

export interface ITextProps {
  as?: ElementType;
  children?: JSX.Element | JSX.Element[] | string;
}

/**
 * Text
 *
 * Use for add text content
 *
 * @returns {JSX.Element}
 */

export const Text = ({ as, children }: ITextProps) => <StyledText as={as}>{children}</StyledText>;
