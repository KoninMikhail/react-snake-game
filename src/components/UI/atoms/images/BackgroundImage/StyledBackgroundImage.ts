import styled from 'styled-components';
import { IBackgroundImageProps } from './BackgroundImage';

export const StyledBackgroundImage = styled.div<IBackgroundImageProps>`
  background-image: url(${({ src }) => src});
  background-position: ${({ position }) => position || 'center center'};
  background-size: ${({ size }) => size || 'cover'};
  width: inherit;
  height: inherit;
`;
