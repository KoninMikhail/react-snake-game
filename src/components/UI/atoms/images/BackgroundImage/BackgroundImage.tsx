import { StyledBackgroundImage } from './StyledBackgroundImage';

/**
 * @typedef IBackgroundImageProps
 *
 * @property {string} src - The image src
 * @property {string=} size - The image size (css rules)
 * @property {string=} position - The image position (css rules)
 * @property {JSX.Element | JSX.Element[]} children - The children elements.
 */
export interface IBackgroundImageProps {
  src: string;
  size?: string;
  position?: string;
  children: JSX.Element | JSX.Element[];
}

/**
 * BackgroundImage
 *
 * Use for add background image
 *
 * @param {IBackgroundImageProps} props- Component properties
 * @returns {JSX.Element}
 */
export const BackgroundImage = ({ children, ...rest }: IBackgroundImageProps): JSX.Element => (
  <StyledBackgroundImage {...rest}>{children}</StyledBackgroundImage>
);
