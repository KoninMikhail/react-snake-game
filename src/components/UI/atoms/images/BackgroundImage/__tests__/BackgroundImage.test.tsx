import { screen, render } from '@testing-library/react';
import { BackgroundImage } from '../BackgroundImage';
import { Text } from '../../../body/Text/Text';

describe('BackgroundImage Image', () => {
  const backgroundImageSrc = '#bgImage.png';
  const backgroundImagePosition = 'center bottom';
  const backgroundImageSize = 'cover';
  const childrenContent = <Text>{'blah'}</Text>;
  const testId = 'background-element';

  it('should be valid css styles', () => {
    render(
      <BackgroundImage
        src={backgroundImageSrc}
        position={backgroundImagePosition}
        size={backgroundImageSize}
        data-testid={testId}
      >
        {childrenContent}
      </BackgroundImage>
    );

    const backgroundElement = screen.getByTestId(testId);

    expect(backgroundElement).toHaveStyleRule('background-image', `url(${backgroundImageSrc})`);
    expect(backgroundElement).toHaveStyleRule('background-position', backgroundImagePosition);
    expect(backgroundElement).toHaveStyleRule('background-size', backgroundImageSize);
  });

  it('should be equal snapshot', () => {
    render(
      <BackgroundImage
        src={backgroundImageSrc}
        position={backgroundImagePosition}
        size={backgroundImageSize}
        data-testid={testId}
      >
        {childrenContent}
      </BackgroundImage>
    );

    expect(screen.getByTestId(testId)).toMatchSnapshot();
  });
});
