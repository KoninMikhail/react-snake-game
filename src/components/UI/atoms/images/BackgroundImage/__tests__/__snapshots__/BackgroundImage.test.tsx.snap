// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage Image should be equal snapshot 1`] = `
.c0 {
  background-image: url(#bgImage.png);
  background-position: center bottom;
  background-size: cover;
  width: inherit;
  height: inherit;
}

.c1 {
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
  color: white;
}

@media (min-width:1200px) {
  .c1 {
    font-size: 1vw;
  }
}

<div
  class="c0"
  data-testid="background-element"
  src="#bgImage.png"
>
  <p
    class="c1"
  >
    blah
  </p>
</div>
`;
